### PR TITLE
feat: Image element tweaks

### PR DESCRIPTION
**Pull request by SHession**
## What does this change?

This PR makes various tweaks to the image element, in order to more closely align it to the existing element.

These changes are:

1. Handling the undefined value of the "weighting" dropdown
2. Displaying a character count for the "caption" field
3. Handling assets with a width and heigh of string 
4. Updates button to use "Guardian Agate Sans"

## How to test

The tests should pass and the behaviour described above should work correctly in the demo and composer.

## How can we measure success?

Our PME image element more closely matches the existing composer version.

## Images

image

## Accessibility

- [ ] [Tested with screen reader](https://github.com/guardian/source/blob/main/docs/06-accessibility.md#screen-readers)
- [x] [Navigable with keyboard](https://github.com/guardian/source/blob/main/docs/06-accessibility.md#keyboard-navigation)
- [ ] [Colour contrast passed](https://github.com/guardian/source/blob/main/docs/06-accessibility.md#colour-contrast)
- [x] [The change doesn't use only colour to convey meaning](https://github.com/guardian/source/blob/main/docs/06-accessibility.md#use-of-colour)
- [x] [Interactive elements show a focus ring when focused by keyboard](https://github.com/guardian/accessibility/blob/main/editorial-tools-guidelines.md#appendix-the-focus-ring)
- [x] [Semantic elements have been used where possible](https://github.com/guardian/accessibility/blob/main/editorial-tools-guidelines.md#appendix-semantic-html)

### CODE REVIEW

File: src/editorial-source-components/InputHeading.tsx [hunk 1]
```diff
@@ -20,12 +21,17 @@ const Errors = ({ errors }: { errors: string[] }) =>
...
-export const InputHeading = ({ label, errors }: Props) => (
+export const InputHeading = ({ label, description, errors }: Props) => (
   <InputHeadingContainer>
     <Label>{label}</Label>
-    <Errors errors={errors} />
+    {errors.length > 0 ? (
+      <Errors errors={errors} />
+    ) : description ? (
+      <Description>{description}</Description>
+    ) : null}
```
Comment (jonathonherbert) [inline src/editorial-source-components/InputHeading.tsx:35]: Ah, I like this approach 👍 not quite convinced these will always be mutually exclusive, but let's find out 😁

File: src/elements/image/ImageElementForm.tsx [hunk 2]
```diff
@@ -145,26 +152,32 @@ const ImageView = ({ field, onChange, errors }: ImageViewProps) => {
     onChange(mediaPayload);
   };
 
-  const getImageSrc = () => {
+  const imageSrc = useMemo(() => {
```
Comment (jonathonherbert) [inline src/elements/image/ImageElementForm.tsx:155]: 👍 nice optimisation

**Review by jonathonherbert — APPROVED** [review]
LGTM. One question – where do we get the inconsistent types (`string`|`number`) from?

One thought – there's a fair amount of weird, domain specific logic building in our transformer for the image element. It might be nice to have tests, if only to document why we need the logic.

**Comment by SHession** [comment]
> LGTM. One question – where do we get the inconsistent types (`string`|`number`) from?
> 
> One thought – there's a fair amount of weird, domain specific logic building in our transformer for the image element. It might be nice to have tests, if only to document why we need the logic.

It's a fair point and something I was a bit uncomfortable with. The fact is that setting it to a string only doesn't do much to reduce the complexity. That being said I could give it another pass to see if I can improve it.

**Review by jonathonherbert — APPROVED** [review]
Tests look good – I think that was the only blocker! Lmk if there's anything else in this PR that needs 👀

**Pull request merged**